libdnf/dnf-context.cpp

```diff
@@ -225,6 +227,8 @@ static void
 dnf_context_init(DnfContext *context)
 {
     DnfContextPrivate *priv = GET_PRIVATE(context);
+    gchar *vars_dir[] = {"/etc/dnf/vars", "/etc/yum/vars"};
```

Comment:
The `g_strdupv` copies NULL-terminated array of strings.  Please add `NULL` terminator into the array:
`{"/etc/dnf/vars", "/etc/yum/vars", NULL};`